### ISSUE TITLE
Killing Frenzy not working

### ISSUE THREAD

**Issue by Obyruz** [lowkey-silvertongue/CH2-Reforged#5]
Just unlock the Killing Frenzy node and after killing a monster I get no buff and no boost to my current haste.

I have tested standing still for more than 5 seconds and then killing a monster to see if my haste would increase afterwards, but there was no change.

This is on the latest Clicker Heroes 2 version 0.10.4

**Comment by lowkey-silvertongue** [comment]
Not sure this has something to do with me. I do not override the default attack function and even if I did, I normally make sure to execute default code were possible before doing something custom.

Stupid question first: does this happen without the mod too?

I changed some things around to make sure I call all default functions and the override code is going to be refactored anyway with the next release. If I find the culprit (if it exists/is on my end) I will fix it.

**Comment by lowkey-silvertongue** [comment]
Okay, seems like it might be on my end; confirmed Killing Frenzy isn't working for me either (ofc this might still be a new bug from the 0.10.x update of the base game, but who knows).
I am going to decompile 0.10 again and look at the skill, maybe I can find out what is going wrong here.

**Comment by Obyruz** [comment]
Fortunately I had a save with Killing Frenzy to test if it's an issue with the mod and after deactivating the mod, Killing Frenzy does work properly on 0.10.4

**Comment by lowkey-silvertongue** [comment]
Yep, I did some testing too, looks like the game code is spaghetti. TL;DR: buffs will not work until I figure out the implementation of the event listener and find a way to manually override it/inject my code. I will peak at some stuff other people have done, maybe someone else found a way around this problem.

Long version:
From what I've gathered, the event listener responsible handles stuff like triggered functions on buffs seperately. I still haven't found the actual implementation, but I hope to find it eventually.
In this case, when a monster is killed, the listener seems to capture that as an event and then call onKilledMonster() and any buff.onKill() functions from all the different buffs registered to the player character at the time. One would believe buff.onKill() would be called inside of onKilledMonster(), but it isn't.
In short, onKilledMonster() is one of several wrapper functions that contain an if clause, and either call onKilledMonsterOverride() when a mod has set the override flag to true or onKilledMonsterDefault() if not. The trigger-function of the Buff-object is not called inside onKilledMonster() or onKilledMonsterDefault(), contrary to what you would expect. Also, pseudo overrides like this make me go yikes, but it's not my game so I have to live with that.

**Comment by Obyruz** [comment]
Sounds very confusing for no reason at all, but why does your mod ends up messing with buffs?

Gave some thought to it, guess you had to override something related to boss killing to be able to allow the skillpoint RNG.
